### PR TITLE
FEATURE: F1-51 Support for data sources in user management

### PR DESCRIPTION
**Pull request by kandl**
- Regenerate api-client-tiger
- Enable paging and filtering for users and groups
- Add support for listing data source identifiers
- Add support for adding and removing data sources to users and groups
- Add support for listing data source permissions
- Add feature flag enableManageDataSourcePermissions

JIRA: F1-51

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/17984/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19519/ : FAILURE in 11m 37s

File: libs/sdk-model/src/dataSources/index.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// (C) 2021-2024 GoodData Corporation
+
+/**
+ * @alpha
+ */
+export type DataSourceType =
```
Comment (scavnickyj) [inline libs/sdk-model/src/dataSources/index.ts:6]: Can we use here some api-client type instead?

Comment (scavnickyj) [reply libs/sdk-model/src/dataSources/index.ts:6]: Ah, this is model, so probably no..

File: libs/sdk-ui-ext/src/internal/components/dialogs/userManagementDialogs/DataSources/AsyncSelectComponents.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    if (isSelectErrorOption(data)) {
+        return ErrorOptionRenderer(data);
+    }
+
+    const sTestStyle = isDataSourceItem(data.value) ? getDataSourceItemTestId(data.value) : "";
+
+    const componentStyle = cx(
+        {
+            option: true,
```
Comment (scavnickyj) [inline libs/sdk-ui-ext/src/internal/components/dialogs/userManagementDialogs/DataSources/AsyncSelectComponents.tsx:88]: What does this do?

File: libs/sdk-ui-ext/src/internal/components/dialogs/userManagementDialogs/DataSources/DataSourceItem.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import { IGrantedDataSource, DataSourcePermissionSubject } from "../types.js";
+
+import { usePermissionsDropdownState } from "./usePermissionsDropdownState.js";
+import { PermissionsDropdown } from "./PermissionsDropdown.js";
+
+const DataSourceIcon: React.FC = () => {
+    return (
+        <div className="gd-grantee-item-icon-left-background">
+            <span className="gd-grantee-item-icon gd-grantee-item-icon-left">
+                <svg
```
Comment (scavnickyj) [inline libs/sdk-ui-ext/src/internal/components/dialogs/userManagementDialogs/DataSources/DataSourceItem.tsx:15]: Would it make sense to move this to a generic ui-kit icon?

File: libs/sdk-backend-tiger/src/backend/dataSources/index.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            return loadAllPages(({ page, size }) =>
+                client.entities
+                    .getAllEntitiesDataSourceIdentifiers({ page, size })
+                    .then((response) => response.data.data.map(convertdataSourceIdentifier)),
+            );
+        });
+    }
+}
+
+async function loadAllPages<T>(
```
Comment (scavnickyj) [inline libs/sdk-backend-tiger/src/backend/dataSources/index.ts:33]: This method is duplicated. Rather move to metadataUtilities.

**Comment by kandl** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/17985/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19522/ : FAILURE in 12m 45s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/17986/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19523/ : SUCCESS in 12m 27s

**Review by scavnickyj — APPROVED** [review]

**Comment by scavnickyj** [comment]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4660

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3880/ : SUCCESS in 6m 46s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19526/ : SUCCESS in 12m 02s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/1535/ : SUCCESS in 9m 52s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/5048/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1759/ : FAILURE in 1m 22s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/1531/ : SUCCESS in 12m 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-compatibility-pipeline/607/ : SUCCESS in 6m 29s

**Pull request merged**